dagmc/dagnav.py

```diff
@@ -500,6 +527,9 @@ def name(self) -> Optional[str]:
 
     @name.setter
     def name(self, val: str):
+        if val.lower() in self.model.group_names:
```

Comment:
The suggestion above about adding a `DAGModel.groups_by_name` method might change how this is approacched.

Reply:
I left this method in since it is slightly cleaner than requesting by name and then taking only the keys.